src/Microsoft.Data.Sqlite/SqliteDataReader.cs

```diff
@@ -94,9 +95,17 @@ public override bool Read()
             }
 
             var rc = NativeMethods.sqlite3_step(_stmt);
-            MarshalEx.ThrowExceptionForRC(rc, _db);
+            try
+            {
+                MarshalEx.ThrowExceptionForRC(rc, _db);
+            }
+            catch
```

Comment:
Is this one needed? If callers put the `DbDataReader` is in a `using` block (like they always should) throwing would cause this (and its statement handle) to be immediately disposed.